File: Backend/mapping-dotnet/MHP.CodingChallenge.Backend.Mapping.Data/DB/ArticleRepository.cs
using System;
using System.Collections.Generic;
using MHP.CodingChallenge.Backend.Mapping.Data.DB.Blocks;

namespace MHP.CodingChallenge.Backend.Mapping.Data.DB
{
    public class ArticleRepository
    { 
        public List<Article> GetAll()
        {
            List<Article> result = new List<Article>();
            result.Add(CreateDummyArticle(1001L));
            result.Add(CreateDummyArticle(2002L));
            result.Add(CreateDummyArticle(3003L));
            result.Add(CreateDummyArticle(4004L));
            result.Add(CreateDummyArticle(5005L));
            return result;
        }

        public Article FindById(long id)
        {
            return CreateDummyArticle(id);
        }

        public void Create(Article article)
        {
            //Ignore
        }

        private Article CreateDummyArticle(long id)
        {
            Article result = new Article();
            result.Id = id;
            result.Author = "Max Mustermann";
            result.Description = "Article Description " + id;
            result.Title = "Article Nr.: " + id;
            result.LastModifiedBy = "Hans Müller";
            result.LastModified = DateTime.Now;
            result.Blocks = CreateBlocks(id);
            return result;
        }

        private HashSet<ArticleBlock> CreateBlocks(long articleId)
        {
            HashSet<ArticleBlock> result = new HashSet<ArticleBlock>();

            TextBlock textBlock = new TextBlock();
            textBlock.Text = "Some Text for " + articleId;
            textBlock.SortIndex = 0;
            result.Add(textBlock);

            ImageBlock imageBlock = new ImageBlock();
            imageBlock.Image = CreateImage(1L);
            textBlock.SortIndex = 1;
            result.Add(imageBlock);

            TextBlock secondTextBlock = new TextBlock();
            secondTextBlock.Text = "Second Text for " + articleId;
            secondTextBlock.SortIndex = 2;
            result.Add(secondTextBlock);

            GalleryBlock galleryBlock = new GalleryBlock();
            secondTextBlock.SortIndex = 3;

            List<Image> galleryImages = new List<Image>();
            galleryImages.Add(CreateImage(2L));
            galleryImages.Add(CreateImage(3L));
            galleryBlock.Images = galleryImages;

            result.Add(galleryBlock);

            TextBlock thirdTextBlock = new TextBlock();
            thirdTextBlock.Text = "Third Text for " + articleId;
            thirdTextBlock.SortIndex = 4;
            result.Add(thirdTextBlock);

            VideoBlock videoBlock = new VideoBlock();
            videoBlock.Type = VideoBlockType.YOUTUBE;
            videoBlock.Url = "https://youtu.be/myvideo";
            videoBlock.SortIndex = 5;

            result.Add(videoBlock);

            return result;
        }

        private Image CreateImage(long imageId)
        {
            Image result = new Image();
            result.Id = imageId;
            result.LastModified = DateTime.Now;
            result.LastModifiedBy = "Max Mustermann";
            result.ImageSize = ImageSize.LARGE;
            result.Url = "https://someurl.com/image/" + imageId;
            return null;
        }
    }
}

File: src/main/kotlin/com/mhp/coding/challenges/mapping/repositories/ArticleRepository.kt
package com.mhp.coding.challenges.mapping.repositories

import com.mhp.coding.challenges.mapping.models.db.Article
import com.mhp.coding.challenges.mapping.models.db.Image
import com.mhp.coding.challenges.mapping.models.db.ImageSize
import com.mhp.coding.challenges.mapping.models.db.blocks.*
import org.springframework.stereotype.Component
import java.util.*

object ArticleRepository {
    fun all(): List<Article> = setOf(1001L, 2002L, 3003L, 4004L, 5005L).map { it.createDummyArticle }

    fun findBy(id: Long): Article = id.createDummyArticle

    fun create(article: Article?) {
        //Ignore
    }

    private val Long.createDummyArticle
        get() = Article(
            id = this,
            lastModified = Date(),
            lastModifiedBy = "Hans Müller",
            title = "Article Nr.: $this",
            description = "Article Description $this",
            author = "Max Mustermann",
            blocks = dummyArticleBlocks,
        )

    private val Long.dummyArticleBlocks: Set<ArticleBlock> by lazy {
        val textBlock = TextBlock(
            text = "Some Text for $this",
            sortIndex = 0
        )

        val imageBlock = ImageBlock(
            image = createImage(1L),
            sortIndex = 1
        )

        val secondTextBlock = TextBlock(
            text = "Second Text for $this",
            sortIndex = 2
        ).also { textBlock.sortIndex = 1 }

        val galleryBlock = GalleryBlock(
            images = listOf(
                createImage(2L),
                createImage(3L)
            )
        ).also { secondTextBlock.sortIndex = 3 }

        val thirdTextBlock = TextBlock(
            text = "Third Text for $this",
            sortIndex = 4
        )

        val videoBlock = VideoBlock(
            type = VideoBlockType.YOUTUBE,
            url = "https://youtu.be/myvideo",
            sortIndex = 4
        )

        setOf(textBlock, imageBlock, secondTextBlock, galleryBlock, thirdTextBlock, videoBlock)
    }

    private fun createImage(imageId: Long): Image? {
        return Image(
            url = "https://someurl.com/image/$imageId",
            id = imageId,
            imageSize = ImageSize.LARGE,
            lastModified = Date(),
            lastModifiedBy = "John Doe"
        ).let { null }
    }
}
